src/plugins/orangehrmDashboardPlugin/Service/EmployeeTimeAtWorkService.php

```diff
@@ -0,0 +1,287 @@
+                    $totalTime += $this->getTimeDifference($endUTCDateTime, $startUTCDateTime);
+                } elseif ($punchInUtcDateTime < $startUTCDateTime && $punchOutUtcDateTime > $startUTCDateTime) {
+                    $totalTime += $this->getTimeDifference($startUTCDateTime, $punchOutUtcDateTime);
+                } elseif ($punchInUtcDateTime < $endUTCDateTime && $punchOutUtcDateTime > $endUTCDateTime) {
+                    $totalTime += $this->getTimeDifference($punchInUtcDateTime, $endUTCDateTime);
+                } else {
+                    $totalTime += $this->getTimeDifference($punchOutUtcDateTime, $punchInUtcDateTime);
+                }
+            }
+        }
+        return $totalTime;
+    }
+
+    /**
+     * @param DateTime $date
+     * @return array  eg:- array(if monday as first day in config => '2021-12-13', '2021-12-19')
+     */
+    private function getWeekBoundaryForGivenDate(DateTime $date): array
+    {
+        $currentWeekFirstDate = date('Y-m-d', strtotime('monday this week', strtotime($date->format('Y-m-d'))));
+        $configDate = $this->getTimesheetPeriodService()->getTimesheetStartDate() - 1;
+        $startDate = date('Y-m-d', strtotime($currentWeekFirstDate . ' + ' . $configDate . ' days'));
+        $endDate = date('Y-m-d', strtotime($startDate . ' + 6 days'));
+        return [$startDate, $endDate];
+    }
+
+    /**
+     * @param int $empNumber
+     * @param DateTime $currentDateTime
+     * @return array
+     * @throws Exception
+     */
+    private function getDataForCurrentWeek(
+        int $empNumber,
+        DateTime $currentDateTime
+    ): array {
+        list($startDate) = $this->getWeekBoundaryForGivenDate($currentDateTime);
+        $startDateTime = new DateTime($startDate . ' 00:00:00', $currentDateTime->getTimezone());
+        $startUTCDateTime = (clone $startDateTime)->setTimezone(
+            new DateTimeZone(DateTimeHelperService::TIMEZONE_UTC)
+        );
+
+        $counter = 0;
+        $weeklyData = [];
+
+        while ($counter < 7) {
+            $totalTimeForDay = $this->getTotalTimeForGivenDate($empNumber, $startUTCDateTime);
+            $weeklyData[] = [
+                'workDay' => [
+                    'id' => $startDateTime->format('w'),
+                    'day' => $startDateTime->format('D'),
+                    'date' => $this->getDateTimeHelper()->formatDateTimeToYmd($startDateTime),
+                ],
+                'totalTime' => $this->getTimeInHoursAndMinutes($totalTimeForDay),
+            ];
+            $startUTCDateTime = clone $startUTCDateTime;
+            $startUTCDateTime = $startUTCDateTime->add(new DateInterval('P1D'));
+            $startDateTime = clone $startDateTime;
+            $startDateTime = $startDateTime->add(new DateInterval('P1D'));
+
+            $this->totalTimeForWeek = $this->totalTimeForWeek + $totalTimeForDay;
```

Comment:
```suggestion
            $totalTimeForWeek += $totalTimeForDay;
```
```php
    /**
     * @param int $empNumber
     * @param DateTime $currentDateTime
     * @return array
     * @throws Exception
     */
    private function getDataForCurrentWeek(
        int $empNumber,
        DateTime $currentDateTime
    ): array {
        // ...
        $totalTimeForWeek = 0;
        while ($counter < 7) {
            //...
            $totalTimeForWeek +=  $totalTimeForDay;
            $counter++;
        }
        return [$weeklyData, $totalTimeForWeek];
    }
```
if you want to keep `$this->totalTimeForWeek` you can expose only one public method to call from API class.

```php
        list($data, $meta) = $this->getEmployeeTimeAtWorkService()->getTimeAtWork();
        return new EndpointResourceResult(
            ArrayModel::class,
            $data,
            new ParameterBag($meta)
        );
```
```php
    public function getTimeAtWork(/** params */)
    {
        return [
            $this->getEmployeeTimeAtWorkService()->getTimeAtWorkData($empNumber, $currentDateTime),
            $this->getEmployeeTimeAtWorkService()->getTimeAtWorkMetaData($empNumber, $currentDateTime),
    }

    private function getTimeAtWorkData(/** params */) 

    private function getTimeAtWorkMetaData(/** params */) 
```
the reason, why I request to change, other devs don't know they should call `getTimeAtWorkMetaData` after `getTimeAtWorkData` to get correct result.